src/gluonts/model/tft/_network.py

```diff
@@ -0,0 +1,488 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import List, Optional, Tuple
+
+import numpy as np
+import mxnet as nx
+from mxnet import gluon
+from mxnet import init
+from mxnet.gluon import nn, HybridBlock
+
+from gluonts.core.component import validated, DType
+from gluonts.model.common import Tensor
+from gluonts.mx.block.feature import FeatureEmbedder as BaseFeatureEmbedder
+from gluonts.mx.block.quantile_output import QuantileOutput
+from gluonts.support.util import weighted_average
+from ._layers import (
+    GatedResidualNetwork,
+    VariableSelectionNetwork,
+    TemporalFusionEncoder,
+    TemporalFusionDecoder,
+)
+
+
+class FeatureEmbedder(BaseFeatureEmbedder):
+    def hybrid_forward(self, F, features: Tensor) -> List[Tensor]:
+        concat_features = super(FeatureEmbedder, self).hybrid_forward(
+            F, features
+        )
+        if self.__num_features > 1:
+            features = F.split(
+                concat_features, num_outputs=self.__num_features, axis=-1
+            )
+        else:
+            features = [concat_features]
+        return features
+
+
+class FeatureProjector(HybridBlock):
+    """
+    Project a sequence of numerical features.
+
+    Parameters
+    ----------
+    feature_dims
+        dimensions for each numerical feature.
+
+    embedding_dims
+        number of dimensions to embed each numerical feature.
+
+    dtype
+        Data type of the embedded features.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        feature_dims: List[int],
+        embedding_dims: List[int],
+        dtype: DType = np.float32,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        assert (
+            len(feature_dims) > 0
+        ), "Length of `cardinalities` list must be greater than zero"
+        assert len(feature_dims) == len(
+            embedding_dims
+        ), "Length of `embedding_dims` and `embedding_dims` should match"
+        assert all(
+            [c > 0 for c in feature_dims]
```

Comment:
The `[]` are not needed here.

Reply:
fixed in abdc7114efd290505ea64ef943baebb656e7b5d0